third-party/realdds/CMakeLists.txt

```diff
@@ -0,0 +1,131 @@
+   "${REALDDS_INCLUDE_DIR}/topics/image/imagePubSubTypes.h"
+   "${REALDDS_INCLUDE_DIR}/topics/image/image.h"
+   "${REALDDS_INCLUDE_DIR}/topics/image/image-msg.h"
+)
+set( REALDDS_NOTIFICATION_TOPIC_HEADER_FILES
+   "${REALDDS_INCLUDE_DIR}/topics/notification/notificationPubSubTypes.h"
+   "${REALDDS_INCLUDE_DIR}/topics/notification/notification.h"
+   "${REALDDS_INCLUDE_DIR}/topics/notification/notification-msg.h"
+)
+set( REALDDS_TOPICS_HEADER_FILES 
+    ${REALDDS_CONTROL_TOPIC_HEADER_FILES}
+    ${REALDDS_DEVICE_INFO_TOPIC_HEADER_FILES}
+    ${REALDDS_IMAGE_TOPIC_HEADER_FILES}
+    ${REALDDS_NOTIFICATION_TOPIC_HEADER_FILES}
+    "${REALDDS_INCLUDE_DIR}/topics/dds-topics.h"
+)
+
+set( REALDDS_HEADER_FILES
+    ${REALDDS_COMMON_HEADER_FILES}
+    ${REALDDS_CLIENT_HEADER_FILES}
+    ${REALDDS_SERVER_HEADER_FILES}
+    ${REALDDS_TOPICS_HEADER_FILES}
+)
+target_sources( ${PROJECT_NAME} PRIVATE ${REALDDS_HEADER_FILES} )
+
+source_group( "Header Files/topics" FILES ${REALDDS_TOPICS_HEADER_FILES} )
+
+
+# Sources -----------------------------------------------------------------------------------
+#
+set( REALDDS_SRC_DIR "${CMAKE_CURRENT_LIST_DIR}/src" )
+
+set( REALDDS_CONTROL_TOPIC_SOURCE_FILES
+   "${REALDDS_SRC_DIR}/controlPubSubTypes.cpp"
+   "${REALDDS_SRC_DIR}/controlTypeObject.h"
+   "${REALDDS_SRC_DIR}/controlTypeObject.cpp"
+   "${REALDDS_SRC_DIR}/control.cpp"
+)
+set( REALDDS_DEVICE_INFO_TOPIC_SOURCE_FILES
+   "${REALDDS_SRC_DIR}/deviceInfo.cpp"
+   "${REALDDS_SRC_DIR}/device-info-msg.cpp"
+   "${REALDDS_SRC_DIR}/deviceInfoTypeObject.h"
+   "${REALDDS_SRC_DIR}/deviceInfoTypeObject.cpp"
+   "${REALDDS_SRC_DIR}/deviceInfoPubSubTypes.cpp"
+)
+set( REALDDS_IMAGE_TOPIC_SOURCE_FILES
+   "${REALDDS_SRC_DIR}/imagePubSubTypes.cpp"
+   "${REALDDS_SRC_DIR}/image.cpp"
+)
+set( REALDDS_NOTIFICATION_TOPIC_SOURCE_FILES
+   "${REALDDS_SRC_DIR}/notificationPubSubTypes.cpp"
+   "${REALDDS_SRC_DIR}/notificationTypeObject.h"
+   "${REALDDS_SRC_DIR}/notificationTypeObject.cpp"
+   "${REALDDS_SRC_DIR}/notification.cpp"
+)
+set( REALDDS_TOPICS_SOURCE_FILES
+    ${REALDDS_CONTROL_TOPIC_SOURCE_FILES}
+    ${REALDDS_DEVICE_INFO_TOPIC_SOURCE_FILES}
+    ${REALDDS_IMAGE_TOPIC_SOURCE_FILES}
+    ${REALDDS_NOTIFICATION_TOPIC_SOURCE_FILES}
+)
+source_group( "Source Files/topics" FILES ${REALDDS_TOPICS_SOURCE_FILES} )
+
+SET( REALDDS_SOURCE_FILES
+    "${REALDDS_SRC_DIR}/dds-guid.cpp"
+    "${REALDDS_SRC_DIR}/dds-participant.cpp"
+    "${REALDDS_SRC_DIR}/dds-topic.cpp"
+    "${REALDDS_SRC_DIR}/dds-topic-reader.cpp"
+    "${REALDDS_SRC_DIR}/dds-device.cpp"
+    "${REALDDS_SRC_DIR}/dds-device-watcher.cpp"
+    "${REALDDS_SRC_DIR}/dds-device-broadcaster.cpp"
+    "${REALDDS_SRC_DIR}/dds-device-server.cpp"
+    ${REALDDS_TOPICS_SOURCE_FILES}
+)
+target_sources( ${PROJECT_NAME} PRIVATE ${REALDDS_SOURCE_FILES} )
+
+
+# Install -----------------------------------------------------------------------------------
+#
+install( TARGETS realdds EXPORT realsense2Targets)
```

Comment:
Does it mean that it will be a separate library? building `shared_libs" will need it along the `librealsense` library? 
If yes we will need to update the windows installer as well, not sure about Debians and test it's working..

Reply:
I also must say that personally I like it better when we don't have 1 BIG CMakelist containing all files . and 1 flat folder containing all topics files.
When browsing this folder it will be hard to find files.
Before we had a folder and a  `.cmake` file for each topic, with the old structure when adding a new topic you create 1 folder and 1 .cmake file and add an include to it on the parent one.

He you will need to change line if different places of this long CMakelist.
What do you think?

Reply:
> Does it mean that it will be a separate library? building `shared_libs" will need it along the `librealsense` library? If yes we will need to update the windows installer as well, not sure about Debians and test it's working..

I just copied from what you did with the `dds` library.
I got an error without it, and I admit I didn't really read up to understand.

Reply:
> I also must say that personally I like it better when we don't have 1 BIG CMakelist containing all files . and 1 flat folder containing all topics files.

How many times are you gonna go into `realdds` to find files?
Having a lot of .cmake files is also very cumbersome. Trust me, I had to go thru them a LOT.
I also don't see many people using separate .cmake files now that `realdds` is a library... it's not like includes that you automatically know how to access.

I don't know, I say leave it alone for now and improve later if needed.

Reply:
* moved all the generated topic files into src/topics/
* a readme on how to generate will be included there (next PR)
* the makefile now include a glob command and will automatically pick up any files you put in either the include/ or src/ directories, so it's now much simpler